pkgs/development/tools/rust/rustup/default.nix

```diff
@@ -48,7 +48,12 @@ rustPlatform.buildRustPackage rec {
   checkFeatures = [ ];
 
   patches = lib.optionals stdenv.isLinux [
-    (runCommand "0001-dynamically-patchelf-binaries.patch" { CC = stdenv.cc; patchelf = patchelf; libPath = "$ORIGIN/../lib:${libPath}"; } ''
+    (runCommand "0001-dynamically-patchelf-binaries.patch"
+      {
+        CC = stdenv.cc;
+        patchelf = patchelf;
+        libPath = "${libPath}";
+      } ''
       export dynamicLinker=$(cat $CC/nix-support/dynamic-linker)
       substitute ${./0001-dynamically-patchelf-binaries.patch} $out \
```

Comment:
I think 32-bit support never properly worked here, I would leave it to you, if you want to fix this in this pull request or later.

Reply:
I don't think this introduces any new regressions. I removed `$ORIGIN/../lib`, but I replaced it with `--add-rpath`.
Furthermore, I changed this because that `rust-lld` requires `$ORIGIN/../../lib`.

Reply:
No, it's not a new regression.